### ISSUE TITLE
Improve /etc/zfs/zpool.cache management

### ISSUE THREAD

**Issue by Lord-Luck** [zfsonlinux/zfs#2433]
Hey @ all

sorry for my bad engish, but after the update from 0.6.2 to 0.6.3 on my debian wheezy box i have the problem that after every reboot the /etc/zfs/zpool.cache -file disappears and i cant find a solution.

i never had this problem befor. i changed nothing but zfs. in dmesg i cant see any error. every 0.6.3 related file is successfully installed and running

my config:
i have 13 disks encryped with luks. after restarting the system i run a script to encrypt every each one of them. after the encrypt i nomaly use the "zfs mount -a" command and everything was fine. i tried many things nothing helped.

the storage box runs only once a week for backup storage, thats why i shut him down once the backups are stored.

than, the next week i boot him up. after the normal and error free boot the /etc/zfs/zpool.cache -file is still there. than i run the encrypt script. at the end every disk is encryped and the /etc/zfs/zpool.cache -file is still there. then i normaly mount the pool. but at this point the /etc/zfs/zpool.cache -file disappears. i found out, that if i use zpool status right after the encrypt the /etc/zfs/zpool.cache -file disappears also. same when i use the zfs list command. the /etc/zfs/zpool.cache -file disappears on every zfs/zpool related command. if i do nothing i cant access the pool. so i use the zpool import command and then i can access the pool. but do i have to import every time? is this a new feature? or a bug? is this bad in any way for the data integrity? like i sad at the beginning, normaly it worked very fine, no problems at all.

any help would be highly appreciated!

Thank you very much.

**Comment by simonvanderveldt** [comment]
Have there been any changes on this?
Asking because `zpool set cachefile=/etc/zfs/zpool.cache tank` doesn't work for me, the cachefile is created but the setting of tank stays `-`, very strange.
Should I created a separate issue for this?
